SW-2303-show-air-filter-version-and-serial-in-settings

JFSolution opened the pull request:
- order mrb_state dict
- add air filter class
- return air filter serial modelId and model name in debug and about settings page

Commit:
- order mrb_State dict
- add air filter class
- return air filter serial modelId and model name in debug and about settings page

irlaec commented on octoprint_mrbeam/__init__.py:
```diff
@@ -36,6 +36,7 @@
 
 from octoprint_mrbeam.rest_handler.update_handler import UpdateRestHandlerMixin
 from octoprint_mrbeam.util.connectivity_checker import ConnectivityChecker
+from .iobeam.airfilter import airfilter
```
Can we change this from relative to absolute import?

JFSolution replied:
sure

irlaec commented on octoprint_mrbeam/iobeam/airfilter.py:
```diff
@@ -0,0 +1,81 @@
+            }
+        return self._pressure1
+
+    def set_serial(self, serial):
```
Have you considered using a decorator the same way you do with the getter? Something like
```suggestion
    @serial.setter
    def serial(self, serial):
```

JFSolution replied:
I will change to the setter decorator

irlaec commented on tests/iobeam/test_aifilter.py:
```diff
@@ -0,0 +1,94 @@
+from octoprint_mrbeam.iobeam.airfilter import AirFilter
+
+
+def test_model_name():
```
I would suggest to split the unit tests that contain more than one assertion in different tests

JFSolution replied:
But it is all testing the same functionality, thats why I grouped it

irlaec replied:
But we are testing different scenarios (more than one AAA) so that's why I would separate it, like stated [here](https://www.educative.io/blog/unit-testing-best-practices-overview#:~:text=7.%20One%20use%20case%20per%20unit%20test). But maybe it's just a matter of preference :)

irlaec commented on octoprint_mrbeam/iobeam/airfilter.py:
```diff
@@ -0,0 +1,81 @@
+# singleton
+from octoprint_mrbeam.mrb_logger import mrb_logger
```
Please add docstrings to the new methods

JFSolution replied:
Ohh yes I missed that, thanks for the hint.

Commit:
Add doc strings
added feedback from PR
improved data handling

Commit:
define the iobeam alpha version

Commit:
- improve about page to show the current air filter info
- show temperatures of airfilter in debug menu
- move mrb_state into init of plugin

Commit:
split tests
